### ISSUE TITLE
Rethink the caching, targeting, etc behavior

### ISSUE THREAD

**Issue by fluffy-critter** [PlaidWeb/Pushl#17]
The actual run loop has gotten fragile and messy. 

Each feed should simply get the list of entries, plus the list of entries which were in the previous cached version. 

Each entry should simply get the links, xored with the links which were in the previous cached version. 

Each ping should only be sent once.

There probably isn’t really a good reason to be running so many threads. Maybe do one active connection per domain, and use that as the gating mechanism for aiohttp.

Basically I feel like a lot of the guts need to be torn out and replaced now that I have a better idea of what I’m doing.
